vcd/resource_vcd_lb_app_rule_test.go

```diff
@@ -0,0 +1,194 @@
+// +build gateway lb lbAppRule ALL functional
+
+package vcd
+
+import (
+	"fmt"
+	"html/template"
+	"regexp"
+	"strings"
+	"testing"
+
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccVcdLBAppRule(t *testing.T) {
+	// The Script parameter must be sent as multiline string separated by \n characters.
+	// Terraform has a native HEREDOC format for sending raw strings (with newline characters).
+	// This variable is established for easier test comparison and is wrapped into HEREDOC syntax
+	// in the `params` map using type `template.HTML` so that template engine does not
+	// escape HEREDOC syntax <<- characters.
+	MultiLineScript := `acl vmware_page url_beg / vmware redirect location https://www.vmware.com/ ifvmware_page
+acl other_page2 url_beg / other2 redirect location https://www.other2.com/ ifother_page2
+`
+
+	// String map to fill the template
+	var params = StringMap{
+		"Org":              testConfig.VCD.Org,
+		"Vdc":              testConfig.VCD.Vdc,
+		"EdgeGateway":      testConfig.Networking.EdgeGateway,
+		"AppRuleName":      t.Name(),
+		"SingleLineScript": "acl vmware_page url_beg / vmware redirect location https://www.vmware.com/ ifvmware_page",
+		"MultilineScript": template.HTML(`<<-EOT
+` + MultiLineScript + `EOT`),
+		"MultilineFailScript": template.HTML(`<<-EOT
+			acl vmware_page url_beg / vmware redirect location https://www.vmware.com/ ifvmware_page
+			acl other_page2 url_beg / other2 redirect location https://www.other2.com/ ifother_page2
+			acl en req.fhdr(accept-language),language(es;fr;en) -m str en
+			use_backend english if en
+		EOT`),
+		"Tags":     "lb lbAppRule",
+		"SkipTest": "",
+	}
+
+	configText := templateFill(testAccVcdLBAppRule_OneLine, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 0: %s", configText)
+
+	params["FuncName"] = t.Name() + "-step1"
+	configText1 := templateFill(testAccVcdLBAppRule_MultiLine, params)
+	debugPrintf("#[DEBUG] CONFIGURATION for step 1: %s", configText1)
+
+	params["FuncName"] = t.Name() + "-step3"
+	// This test must fail with invalid rule script so we avoid running it in `make test-binary`
+	params["SkipTest"] = "# skip-test"
```

Comment:
The way skip test is implemented, it allows you to add a short message after "skip-test" and until the end of the line, so that you can state the reason for skipping, and `test-binary.sh` will report such reason.
For example: `# skip-test: it will fail on purpose`

Reply:
Very nice. Using it.